pyccel/codegen/printing/ccode.py

```diff
@@ -1856,7 +1856,8 @@ def _print_FunctionCall(self, expr):
          # Ensure the correct syntax is used for pointers
         args = []
         for a, f in zip(expr.args, func.arguments):
-            a = a.value if a else Nil()
+            a = a.value
+            a = a if a else Nil()
```

Comment:
Can you not write this?
```suggestion
            a = a.value or Nil()
```
By the way, we could use a different variable instead of redefining `a`.